### DISCUSSION TITLE
Is there a way to draw rectangle on image with this lib?

### DISCUSSION THREAD

**Discussion by barbaragabriella** [nothings/stb#1360] [Q&A]
I would like to know if there is a way to draw rectangle + label on open image with stb library?

If yes, any example?

Thank you

**Comment by thedmd** [comment]
No, not directly. You can use stb libraries to help build such functionality.

To do such task on CPU you would most likely need:
* [stb_image](https://github.com/nothings/stb/blob/master/stb_image.h) - for image loading
* [stb_image_write](https://github.com/nothings/stb/blob/master/stb_image_write.h) - to save final image
* [stb_truetype.h](https://github.com/nothings/stb/blob/master/stb_truetype.h) - to load TTF and rasterize letters
* Any drawing routines like image copying and drawing rectangles need to be written by hand
